core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,323 @@
+import * as d3 from "d3";
+
+import { Chart } from "../../chart";
+
+import "./horizontalBarChart.css";
+const initializingBarHeight = 2;
+const margin = {
+  top: 10,
+  right: 0,
+  bottom: 20,
+  left: 0,
+};
+const barValueMargin = 4;
+const barPadding = 0.2;
+const groupHeadlineHeight = 14;
+const headlinePadding = 6;
+const barTextBottomMargin = 6;
+/**
+ * Visualizes data as a cluster of bubbles where the value of the bubble is represented as the radius.
+ *
+ * The bubbles are being added in a spiral starting in the center of the cluster meaning sorted data will lead to all small bubbles in the middle or outside.
+ *
+ * @class
+ * @extends Chart
+ * @param {CSS-selector} selector - A CSS selector, where the svg will be attached to
+ * @param {Object[]} data - The data to be visualized as a bubble cluster
+ * @param {string} data[].title - The title/name the bubble has
+ * @param {number} data[].value - The value of the bubble, which corresponds to it's radius
+ * @param {number = 400} [width] - The width of the svg
+ * @param {number = 200} [height] - The height of the svg
+ * @param {string|callback = "blue"} [barColor] - The color of the bar (callbacks receive event and data)
+ * @param {string = null} [barValueColor] - The color the values are shown in (default = no values shown)
+ * @param {number = 4} [numberTicksY] - Number of Ticks on the y-axis (will deviate by 1 if the values wouldn't make a nice scale otherwise)
+ */
+export class HorizontalBarChart extends Chart {
+  constructor({
+    selector,
+    data,
+    barColor = "blue",
+    width = 400,
+    height = 400,
+    barValueColor,
+    //barWidth is actually bar height since it is a horizontal bar chart but for all intents and purposes it is the bar's width
+    barWidth,
+    numberTicksX,
+    groups,
+  }) {
+    super({ selector, data, width, height, margin });
+    this._groups = groups;
+    this._barColor = barColor || "blue";
+    this._barWidth = barWidth;
+    this._xScale = d3.scaleLinear().range([0, this.chartWidth]);
+    this._yScales = this._getYscales(groups);
+    this._barValueColor = barValueColor;
+    this._numberTicksX = numberTicksX || 4;
+  }
+
+  _getYscales(groups) {
+    if (groups) {
+      const yScales = [];
+      let headingSpace = groupHeadlineHeight + headlinePadding;
+      for (let group of groups) {
+        const relevantBars = this.data.filter(
+          (data) => data.group === group.id
+        ).length;
+        // totalSpan * barPadding = paddingSpan
+        const barsSpan = relevantBars * this._barWidth;
+        const paddingSpan = barsSpan / (1 / barPadding - 1);
+        const totalSpan = barsSpan + relevantBars * paddingSpan;
+        // const totalSpan = barsSpan;
+        const scale = d3
+          .scaleBand()
+          .range([headingSpace, totalSpan])
+          .padding(barPadding);
+        headingSpace = headingSpace + totalSpan;
+        yScales.push({ scale, id: group.id });
```

Comment:
OK, I see the origin of `scale.scale` below. IIRC we're calling this kind of thing `enhanced` or `extended` but in this case it's simply scale-with-id. scaleWithId. Maybe it could be called that way below... Also, not totally clear why this is needed, but if we're going to reuse it a bit more extensively, we could define a class and document it; it would make it easier to refactor and generally understand.

Reply:
I see, yes the reason why this is need is when the bar chart is generated depending on the data, the grouped data needs to identify which scale it is supposed to go into as each scale has different positioning and range. I do agree this can be a class called ScaleContainer = { scale : scale, id: string}  which will help keeping things clear and if anything else is needed later we can include easily.